src/browser.qml

```diff
@@ -48,4 +86,45 @@ ApplicationWindow {
         property: "mainWindow"
         value: webView
     }
+
+    Component {
+        id: backgroundComponent
+        Browser.Background {
+            id: bg
+            property Item pageContainer
+            property Page page
+
+            // Page and attached background is destroyed when the page gets destroyed.
+            onPageChanged: {
+                if (!page) {
+                    var index = 0
+                    for (; index < backgrounds.length; ++index) {
+                        if (backgrounds[index].background === bg) {
+                            break
+                        }
+                    }
+
+                    if (index === 0 && backgrounds.length > 0) {
```

Comment:
I guess that given `index` is initialized to -1 these three conditions could be collapsed to only one
```
if (index >=  0 && backgrounds.length > 0) {
    background.splice(index, 1)
} else {
    backgrounds = []
}
```